### PR TITLE
improve wsl guide clarity

### PR DESCRIPTION
**Pull request by enclave-marc-barry**

### CODE REVIEW

File: docs/tutorials/deploying-gateways-in-wsl.md [hunk 1]
```diff
@@ -1,46 +1,98 @@
...
+
+1. Nested virtualiation introduces an additional layer of abstraction, which can lead to reduced compute performance compared to native virtualisation, which can negativley affect throughput.
+
+2. By default, WSL uses a type of Network Address Translation (NAT) which prevents Enclave from establishing peer-to-peer connections, also causing reduced network performance as traffic passes instead through relay servers.
+
+As a result, we don't recommend runing an Enclave Gateway inside WSL due to the complication inroduced by nested virtualisation and WSL NAT, unless no other alternatives are available.
+
+If you decide to run an Enclave Gateway using nested virtualisation, this guide will help you achieve the best performance for your Gateway.
+
+The following steps show you how to enable port forwarding and force an Enclave Gateway to run on a fixed port number as a work-around to enable direct peer-to-peer connections. Please note however, the Enclave Gateway in WSL will only support TCP connections, which may still result in reduced network performance compared to native virtualisation, or no virtualisation at all.
```
Comment (enclave-charlie) [inline docs/tutorials/deploying-gateways-in-wsl.md:28]: I'd shove "However" in front of "Please note" to reduce the number of commas required:

"However, please note that the Enclave Gateway in WSL will only support TCP connections, which may still result in reduced network performance compared to native virtualisation, or no virtualisation at all."

File: docs/tutorials/deploying-gateways-in-wsl.md [hunk 2]
```diff
@@ -1,46 +1,98 @@
...
+    netsh interface portproxy add v4tov4 listenaddress=0.0.0.0 listenport=59724 connectport=59724 connectaddress=[ipaddr]
+    ```
+
+5. Open the Windows Firewall by running the following `netsh` command on the Windows host machine.
+
+    ```shell
+    netsh advfirewall firewall add rule name="Enclave WSL2" dir=in action=allow protocol=TCP localport=59724
+    ``` 
+
+6. Normally Enclave selects a random port number to use each time it's started, but you'll need to fix Enclave to use the same port number you've forwared to WSL on the Windows host. You can force Enclave to use a fixed port number by running the following command inside your WSL instance:
```
Comment (enclave-charlie) [inline docs/tutorials/deploying-gateways-in-wsl.md:63]: "Forwarded" typo

File: docs/tutorials/deploying-gateways-in-wsl.md [hunk 3]
```diff
@@ -1,46 +1,98 @@
...
+
+    ```bash
+    enclave set-config localport 59724
+    ```
+
+    You may need to restart Enclave for the change to take effect, if you’re prompted to restart Enclave, do so.
```
Comment (enclave-charlie) [inline docs/tutorials/deploying-gateways-in-wsl.md:69]: Split into two sentences:

"You may need to restart Enclave for the change to take effect. If you’re prompted to restart Enclave, do so."

File: docs/tutorials/deploying-gateways-in-wsl.md [hunk 4]
```diff
@@ -1,46 +1,98 @@
...
+        enclave set-config localport 59724 --profile Universe
+        ```
+
+7. You can confirm that Enclave is using the new fixed port by running the following command inside your WSL instance:
+
+    ```bash
+    enclave status
+    ```
+    
+    The port number in the `Binding address` should match the value you chose for port forwarding, which is `59724` if you use the same port number we did in this guide.
```
Comment (enclave-charlie) [inline docs/tutorials/deploying-gateways-in-wsl.md:83]: "if you've used the same port number as we did in this guide"?

File: docs/tutorials/deploying-gateways-in-wsl.md [hunk 5]
```diff
@@ -1,46 +1,98 @@
...
-We don't recommend that users run a gateway inside WSL unless there are no alternatives available. Additional configuration is required, and the gateway will only support TCP connections.
+However, in some scenarios customers may not have easy access to Linux hosts. In such cases, the Windows Susbsystem for Linux (WSL) allows Linux distributions to be hosted directly from within Windows using a concept known as [nested virtualisation](https://learn.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/nested-virtualization) (running a VM inside a VM).
+
+WSL and nested virtualisation enable organisations which don't, or can't, run Linux servers to avoid the overhead of provisioning additional dedicated resources in order to run an Enclave Gateway. However, there are limitations to be aware of when deploying an Enclave Gateway inside WSL.
+
+## Limitations
+
+When running Enclave inside WSL there are two limitations to be aware of:
+
+1. Nested virtualiation introduces an additional layer of abstraction, which can lead to reduced compute performance compared to native virtualisation, which can negativley affect throughput.
```
Comment (enclave-charlie) [inline docs/tutorials/deploying-gateways-in-wsl.md:20]: "negatively" typo

**Review by enclave-charlie — CHANGES_REQUESTED** [review]
Thanks for the changes, Marc - this is substantially improved. 

Do we need to change the title of the document in the index? It's still "Deploying Gateways in WSL".

File: docs/tutorials/deploying-gateways-in-wsl.md (hunk 1 above)
Comment (enclave-marc-barry) [reply docs/tutorials/deploying-gateways-in-wsl.md:28]: Fixed, ty

File: docs/tutorials/deploying-gateways-in-wsl.md (hunk 2 above)
Comment (enclave-marc-barry) [reply docs/tutorials/deploying-gateways-in-wsl.md:63]: I need to get a vscode spell check extension, good spot - thank you

File: docs/tutorials/deploying-gateways-in-wsl.md (hunk 4 above)
Comment (enclave-marc-barry) [reply docs/tutorials/deploying-gateways-in-wsl.md:83]: agreed, that's better. fixed

File: docs/tutorials/deploying-gateways-in-wsl.md (hunk 5 above)
Comment (enclave-marc-barry) [reply docs/tutorials/deploying-gateways-in-wsl.md:20]: ty ty

**Comment by enclave-marc-barry** [comment]
Good question about changing the title. Tom has already shared the URL with a partner - so if we do, we just need to create a redirect (if it changes the URL). What do you think? In the mean time I'll push up the fixes you flagged

**Comment by enclave-charlie** [comment]
@enclave-marc-barry,  We can change the title of the document on the tutorials page without changing the url. That might be the easiest option.

File: docs/tutorials/deploying-gateways-in-wsl.md [hunk 6]
```diff
@@ -17,15 +17,15 @@ WSL and nested virtualisation enable organisations which don't, or can't, run Li
...
 
 As a result, we don't recommend runing an Enclave Gateway inside WSL due to the complication inroduced by nested virtualisation and WSL NAT, unless no other alternatives are available.
 
 If you decide to run an Enclave Gateway using nested virtualisation, this guide will help you achieve the best performance for your Gateway.
 
-The following steps show you how to enable port forwarding and force an Enclave Gateway to run on a fixed port number as a work-around to enable direct peer-to-peer connections. Please note however, the Enclave Gateway in WSL will only support TCP connections, which may still result in reduced network performance compared to native virtualisation, or no virtualisation at all.
+The following steps show you how to enable port forwarding and force an Enclave Gateway to run on a fixed port number as a work-around to enable direct peer-to-peer connections. However, please note the Enclave Gateway in WSL will only support TCP connections, which may still result in reduced network performance compared to native virtualisation, or no virtualisation at all.
```
Comment (enclave-charlie) [inline docs/tutorials/deploying-gateways-in-wsl.md:28]: "please note that"?

File: docs/tutorials/index.md [hunk 7]
```diff
@@ -50,9 +50,9 @@ hide: navigation, toc
   image: ../images/guides/thumbnails/guacamole.png
   url: ./using-guacamole-with-enclave.md
 
-- title: Deploying Gateways in WSL
+- title: Deploying an Enclave Gateway inside WSL
```
Comment (enclave-charlie) [inline docs/tutorials/index.md:53]: Capitalise "inside"

File: docs/tutorials/index.md [hunk 8]
```diff
@@ -50,9 +50,9 @@ hide: navigation, toc
...
   content: |
-    Deploying Gateways in WSL
+    Deploying an Enclave Gateway inside WSL
```
Comment (enclave-charlie) [inline docs/tutorials/index.md:55]: Capitalise "inside"

**Review by enclave-charlie — CHANGES_REQUESTED** [review]

**Comment by enclave-charlie** [comment]
The longer title looks a bit weird:

image

Could change "inside" to "in"?

**Review by enclave-charlie — APPROVED** [review]
Looks good 👍

**Pull request merged**